src/flat-button.jsx

```diff
@@ -109,6 +111,20 @@ let FlatButton = React.createClass({
     }, labelStyle);
 
     let labelElement = label ? <span style={mergedLabelStyles}>{label}</span> : null;
+    let children;
+
+    // Place label before or after this.props.children.
+    if (this.props.labelPosition === 'right') {
+      children = React.addons.createFragment({
+        left: this.props.children,
+        right: labelElement,
+      });
```

Comment:
This change will reverse the order by default. I think we need to keep the original order so there aren't any unexpected results.

You can see that originally, the label came before children - https://github.com/callemall/material-ui/pull/1286/files#diff-25151f62042b3bd6779c91508a11cb56L124

Now by default, the label comes after children.